File: service_edu/router/teacher.go
package router

import (
	"github.com/gin-gonic/gin"
	"github.com/thisIsCaiJi/online_college/service_edu/controller"
)

func teacherRouter(group *gin.RouterGroup){
	group.POST("/pageTeacherCondition/:current/:limit",controller.PageTeacherConditionPost)

	group.POST("/addTeacher",controller.AddTeacher)

	group.GET("/getTeacher/:id",controller.GetTeacherById)

	group.POST("/updateTeacher",controller.UpdateTeacher)

	group.DELETE("/:id",controller.RemoveTeacher)
}

File: online-college-vue/src/api/edu/teacher.js
import request from '@/utils/request'

export default{
    getTeacherListPage(current,limit,teacherQuery) {
        return request({
          url: `/eduservice/edu-teacher/pageTeacherCondition/${current}/${limit}`,
          method: 'post',
          data: teacherQuery
        })
    },
    removeTeacherById(id) {
      return request({
        url: `/eduservice/edu-teacher/${id}`,
        method: 'delete'
      })
    },
    addTeacher(teacher) {
      return request({
        url: `/eduservice/edu-teacher/addTeacher`,
        method: 'post',
        data: teacher
      })
    },
    getTeacherById(id) {
      return request({
        url: `/eduservice/edu-teacher/getTeacher/${id}`,
        method: 'get'
      })
    },
    updateTeacher(teacher) {
      return request({
        url: `/eduservice/edu-teacher/updateTeacher`,
        method: 'post',
        data: teacher
      })
    }
}
